ambry-network/src/main/java/com.github.ambry.network/NetworkClient.java

```diff
@@ -138,17 +153,22 @@ public NetworkClient(Selector selector, NetworkConfig networkConfig, int maxConn
             connId = selector.connect(new InetSocketAddress(host, port.getPort()), networkConfig.socketSendBufferBytes,
                 networkConfig.socketReceiveBufferBytes, port.getPortType());
             connectionTracker.startTrackingInitiatedConnection(host, port, connId);
+            requestMetadata.connectionCheckOutAttempts++;
           }
         } else {
-          sends.add(new NetworkSend(connId, requestMetadata.requestInfo.getRequest(), null, time));
+          sends.add(new NetworkSend(connId, requestMetadata.requestInfo.getRequest(),
+              requestMetadata.clientNetworkRequestMetrics, time));
           connectionIdToRequestInFlight.put(connId, requestMetadata);
           iter.remove();
+          numPendingConnections.set(pendingRequests.size());
```

Comment:
Actually I think this is a non-issue. There is no problem with this particular metric actually, as the gauges are per `NetworkClient`. Anything that is not a gauge is common to all, and get aggregated across the `NetworkClient`s, which is probably what you want. But do ensure that.